### DISCUSSION TITLE
Request:  Ashlar to make additional files like Montage when limit reached

### DISCUSSION THREAD

**Discussion by radialmonster** [ImageMagick/ImageMagick#5099] [Development] [answered]
In Montage you can set a row and column and once you reach that limit, it will create a new file for additional pictures, and keep creating new files as needed.

I'd like Ashlar to do the same sort of thing.  I'd like to limit the number of images that can be in the Ashlar.  You would not set the row and column as Ashlar will determine that automatically.  Once it reaches the number set it will create another file with the next set.

This would allow it to see a large list of images, and figure out the best layout to fit the image limit specified for each ashlar.

Currently I have a script that separates a large directory of images into separate folders with each folder having 4 images, then i run ashlar on those 4 images only.  Then it moves to the next folder and runs another ashlar on those 4 images etc.

This works, but the layouts could be more optimized to reduce blank spaces if ashlar could analyze a large number of images and layout multiple outputs all optimized. 

thanks!

**Comment by urban-warrior** [comment, marked as answer]
We added a new define to address your issue.  Here is the new ASHLAR description:

> By default, a reasonable canvas size and border width is determined relative to the image collection you provide. You can explicitedly set the canvas size and border width by appending to the filename, e.g. ashlar:canvas.png[1024x768+4+4]. By default, alignment is along the left edge. Use -define ashlar:best-fit=true to align on both the left and right edges. You can label the image tiles with, for example, -label %f. By default, all image tiles are rendered on a single canvas. Use -define ashlar:tiles=50, for example, to render at most 50 tiles per canvas.

**Reply by radialmonster** [reply]
That idea works too!  I will give it a try thank you!!!

**Reply by urban-warrior** [reply]
The define is only available in the main repo for now.  You need to clone the main repo to test the new feature.

**Reply by radialmonster** [reply]
ah I see.  I will have to wait for windows build then.  Its ok.  Your idea will work just fine though. thank you!

**Comment by fmw42** [comment]
@urban-warrior 

> Use -define ashlar:tiles=50, for example, to render at most 50 tiles per canvas.

What release was this introduced?  Is this documented where I can find it easily in the change log?

**Reply by urban-warrior** [reply]
It was introduced this morning.  The formats page mentions the new define.

**Reply by fmw42** [reply]
Thanks for that. But if it is not showing in the change log in a form that I can scan and find, then it does not help me know in which release it was introduced.

**Reply by fmw42** [reply]
P.S. I do not see it listed on the defines page.  Have I missed it?  Have you not posted it yet?

**Reply by fmw42** [reply]
All of the defines for ashlar are missing on the defines page. Please add them.

**Reply by urban-warrior** [reply]
They are on the defines page.  They will show up once the cache for that page expires later today.
